include/circt/Dialect/SV/SVStatements.td

```diff
@@ -138,6 +138,60 @@ def AlwaysOp : SVOp<"always", [HasRegionTerminator, NoRegionArguments,
   let verifier = "return ::verifyAlwaysOp(*this);";
 }
 
+def NoReset: I32EnumAttrCase<"NoReset", 0, "noreset">;
+def SyncReset: I32EnumAttrCase<"SyncReset", 1, "syncreset">;
+def AsyncReset: I32EnumAttrCase<"AsyncReset", 2, "asyncreset">;
+
+def ResetTypeAttr : I32EnumAttr<"ResetType", "reset type",
+                                   [NoReset, SyncReset, AsyncReset]>;
+
+
+def AlwaysFFOp : SVOp<"alwaysff", [HasRegionTerminator, NoRegionArguments,
+                               RecursiveSideEffects]> {
+  let summary = "'alwaysff @' block with optional reset";
+  let description = [{ 
+    alwaysff blocks represent always_ff verilog nodes, which enforce inference 
+    of registers.  This block takes a clock signal and edge sensitivity and 
+    reset type.  If the reset type is anything but 'noreset', the block takes a
+    reset signal, reset sensitivity, and reset block.  Appropriate if conditions
+    are generated in the output code based on the reset type.  A negative-edge, 
+    asynchronous reset will check the inverse of the reset condition 
+    (if (!reset) begin resetblock end) to match the sensitivity.
+  }];
+  
+  let regions = (region SizedRegion<1>:$bodyBlk, AnyRegion:$resetBlk);
+  let arguments = (ins EventControlAttr:$clockEdge, I1:$clock, 
+                       ResetTypeAttr:$resetStyle, 
```

Comment:
Shouldn't $resetStyle be optional as well? I would suggest that `noreset` should be the default.

Reply:
The attribute exists regardless, the builder for noreset sets it implicitly.  So here, it is required.  In the assembly format it could be optional.

Reply:
Optional in the assembly format seems better to me.